ouroboros-network/src/Ouroboros/Network/Server/Version.hs

```diff
@@ -0,0 +1,133 @@
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE TypeOperators #-}
+{-# LANGUAGE ConstraintKinds #-}
+
+module Ouroboros.Network.Server.Version
```

Comment:
This is totally fine for now, but we will eventually want to replace it (incorporating good ideas) with the intended stuff for the "mux control" version negotiation & network interface @coot is working on. The proxy and download clients should both be able to use this library interface.